src/Shared/Components/CICDHistory/DeploymentDetailSteps.tsx

```diff
@@ -0,0 +1,165 @@
+import { useEffect, useRef, useState } from 'react'
+import { useHistory, useParams, useRouteMatch } from 'react-router'
+import {
+    DEPLOYMENT_STATUS,
+    DeploymentAppTypes,
+    GenericEmptyState,
+    Progressing,
+    TIMELINE_STATUS,
+    URLS,
+} from '../../../Common'
+import { getDeploymentStatusDetail } from './service'
+import {
+    DeploymentDetailStepsType,
+    DeploymentStatusDetailsBreakdownDataType,
+    DeploymentStatusDetailsType,
+} from './types'
+import { processDeploymentStatusDetailsData } from './utils'
+import { DEPLOYMENT_STATUS_QUERY_PARAM, EMPTY_STATE_STATUS } from './constants'
+import { ReactComponent as Arrow } from '../../../Assets/Icon/ic-arrow-forward.svg'
+import mechanicalOperation from '../../../Assets/Icon/ic-mechanical-operation.svg'
+import CDEmptyState from './CDEmptyState'
+import DeploymentStatusDetailBreakdown from './DeploymentStatusBreakdown'
+
+let deploymentStatusTimer = null
+const DeploymentDetailSteps = ({
+    deploymentStatus,
+    deploymentAppType,
+    isHelmApps = false,
+    installedAppVersionHistoryId,
+    isGitops,
+    userApprovalMetadata,
+    isVirtualEnvironment,
+    processVirtualEnvironmentDeploymentData,
+    renderDeploymentApprovalInfo,
+}: DeploymentDetailStepsType) => {
+    const history = useHistory()
+    const { url } = useRouteMatch()
+    const { appId, envId, triggerId } = useParams<{ appId: string; envId?: string; triggerId?: string }>()
+    const [deploymentListLoader, setDeploymentListLoader] = useState<boolean>(
+        deploymentStatus?.toUpperCase() !== TIMELINE_STATUS.ABORTED,
+    )
+    const isVirtualEnv = useRef(isVirtualEnvironment)
+    const processedData = isVirtualEnv.current
+        ? processVirtualEnvironmentDeploymentData()
```

Comment:
please add check of existence of processVirtualEnvironmentDeploymentData. Same for renderDeploymentApprovalInfo across

Reply:
was previously imported in dashboard from FE Lib, so check was required, now they are passed as props, no need to check for existence

Reply:
Even then it required